python/gtsam/examples/TranslationAveragingExample.py

```diff
@@ -0,0 +1,126 @@
+"""
+GTSAM Copyright 2010-2018, Georgia Tech Research Corporation,
+Atlanta, Georgia 30332-0415
+All Rights Reserved
+Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+See LICENSE for the license information
+
+This example shows how 1dsfm uses outlier rejection (MFAS) and optimization (translation recovery)
+together for estimating global translations from relative translation directions and global rotations.
+The purpose of this example is to illustrate the connection between these two classes using a small SfM dataset.
+
+Author: Akshay Krishnan
+Date: September 2020
+"""
+
+from collections import defaultdict
+from typing import Tuple, List
+
+import numpy as np
+
+import gtsam
+from gtsam.examples import SFMdata
+
+# Hyperparameters for 1dsfm, values used from Kyle Wilson's code.
+MAX_1DSFM_PROJECTION_DIRECTIONS = 48
+OUTLIER_WEIGHT_THRESHOLD = 0.1
+
+
+def get_data() -> Tuple[gtsam.Values, List[gtsam.BinaryMeasurementUnit3]]:
+    """"Returns global rotations and unit translation directions between 8 cameras 
+    that lie on a circle and face the center. The poses of 8 cameras are obtained from SFMdata 
+    and the unit translations directions between some camera pairs are computed from their 
+    global translations. """
+    # Using toy dataset in SfMdata for example.
+    wTc = SFMdata.createPoses(gtsam.Cal3_S2(50.0, 50.0, 0.0, 50.0, 50.0))
+    # Rotations of the cameras in the world frame.
+    wRc_values = gtsam.Values()
+    # Normalized translation directions from camera i to camera j
+    # in the coordinate frame of camera i.
+    i_iZj_list = []
+    for i in range(0, len(wTc) - 2):
+        # Add the rotation.
+        wRc_values.insert(i, wTc[i].rotation())
+        # Create unit translation measurements with next two poses.
+        for j in range(i + 1, i + 3):
+            i_iZj = gtsam.Unit3(wTc[i].rotation().unrotate(
+                wTc[j].translation() - wTc[i].translation()))
+            i_iZj_list.append(gtsam.BinaryMeasurementUnit3(
+                i, j, i_iZj, gtsam.noiseModel.Isotropic.Sigma(3, 0.01)))
+    # Add the last two rotations.
+    wRc_values.insert(len(wTc) - 1, wTc[-1].rotation())
+    wRc_values.insert(len(wTc) - 2, wTc[-2].rotation())
+    return (wRc_values, i_iZj_list)
+
+
+def estimate_poses(i_iZj_list: gtsam.BinaryMeasurementsUnit3,
+                   wRc_values: gtsam.Values) -> gtsam.Values:
+    """Estimate poses given rotations and normalized translation directions between cameras.
+
+    Args:
+        iZj_list -- List of normalized translation direction measurements between camera pairs, 
```

Comment:
i think this should say
```python
Args:
     i_iZj_list - ...
````

Reply:
Also, these should be colons here, not hyphens (`argname: ...`, not `argname -- ...` (https://google.github.io/styleguide/pyguide.html#383-functions-and-methods):

```python
        wRc_values: Rotations of the cameras in the world frame.
    Returns:
        Values: Estimated poses.
```

Reply:
Okay, I am not very familiar with Python conventions, was just following Shonan Averaging. https://github.com/borglab/gtsam/blob/develop/python/gtsam/examples/ShonanAveragingCLI.py